From 7da66eea9f68e4abc83ed2892114ec565eddd66a Mon Sep 17 00:00:00 2001
From: Nathan Herald <me@nathanherald.com>
Date: Wed, 31 Oct 2018 09:03:56 -0700
Subject: [PATCH] debian has iptables-legacy and iptables-nft now
Origin: upstream, https://github.com/docker/libnetwork/pull/2285

Signed-off-by: Nathan Herald <me@nathanherald.com>
---
 iptables/iptables.go | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/iptables/iptables.go b/iptables/iptables.go
index 4b8d8832e9..6a108e93d0 100644
--- a/libnetwork/iptables/iptables.go
+++ b/libnetwork/iptables/iptables.go
@@ -87,11 +87,16 @@ func initFirewalld() {
 }
 
 func detectIptables() {
-	path, err := exec.LookPath("iptables")
+	path, err := exec.LookPath("iptables-legacy") // debian has iptables-legacy and iptables-nft now
 	if err != nil {
-		return
+		path, err = exec.LookPath("iptables")
+		if err != nil {
+			return
+		}
 	}
+
 	iptablesPath = path
+
 	supportsXlock = exec.Command(iptablesPath, "--wait", "-L", "-n").Run() == nil
 	mj, mn, mc, err := GetVersion()
 	if err != nil {
