Index: docker.io/pkg/archive/changes_test.go
===================================================================
--- docker.io.orig/pkg/archive/changes_test.go
+++ docker.io/pkg/archive/changes_test.go
@@ -197,6 +197,7 @@ func TestChangesWithChanges(t *testing.T
 
 // See https://github.com/docker/docker/pull/13590
 func TestChangesWithChangesGH13590(t *testing.T) {
+	t.Skip("DM - skipped, doesn't work under pbuilder")
 	// TODO Windows. There may be a way of running this, but turning off for now
 	// as createSampleDir uses symlinks.
 	if runtime.GOOS == "windows" {
