--- a/components/engine/contrib/init/systemd/docker.service
+++ b/components/engine/contrib/init/systemd/docker.service
@@ -10,9 +10,9 @@
 # the default is not to use systemd for cgroups because the delegate issues still
 # exists and systemd currently does not support the cgroup feature set required
 # for containers run by docker
 EnvironmentFile=-/etc/default/docker
-ExecStart=/usr/bin/dockerd -H fd:// $DOCKER_OPTS
+ExecStart=/usr/sbin/dockerd -H fd:// $DOCKER_OPTS
 ExecReload=/bin/kill -s HUP $MAINPID
 LimitNOFILE=1048576
 # Having non-zero Limit*s causes performance problems due to accounting overhead
 # in the kernel. We recommend using cgroups to do container-local accounting.
--- a/components/engine/contrib/init/sysvinit-debian/docker
+++ b/components/engine/contrib/init/sysvinit-debian/docker
@@ -21,9 +21,9 @@
 
 BASE=docker
 
 # modify these in /etc/default/$BASE (/etc/default/docker)
-DOCKERD=/usr/bin/dockerd
+DOCKERD=/usr/sbin/dockerd
 # This is the pid file managed by docker itself
 DOCKER_PIDFILE=/var/run/$BASE.pid
 # This is the pid file created/managed by start-stop-daemon
 DOCKER_SSD_PIDFILE=/var/run/$BASE-ssd.pid
--- a/components/engine/contrib/init/upstart/docker.conf
+++ b/components/engine/contrib/init/upstart/docker.conf
@@ -14,9 +14,9 @@
 kill timeout 20
 
 script
 	# modify these in /etc/default/$UPSTART_JOB (/etc/default/docker)
-	DOCKERD=/usr/bin/dockerd
+	DOCKERD=/usr/sbin/dockerd
 	DOCKER_OPTS=
 	if [ -f /etc/default/$UPSTART_JOB ]; then
 		. /etc/default/$UPSTART_JOB
 	fi
