Description: allow skipping "privileged" tests with "-test.short"
Author: Tianon Gravi <tianon@debian.org>
Forwarded: no

--- a/engine/builder/dockerfile/evaluator_test.go
+++ b/engine/builder/dockerfile/evaluator_test.go
@@ -93,8 +93,9 @@
 	return dispatchTestCases
 }
 
 func TestDispatch(t *testing.T) {
+	t.Skip("DM - skipping privileged test")
 	testCases := initDispatchTestCases()
 
 	for _, testCase := range testCases {
 		executeTestCase(t, testCase)
--- a/engine/builder/dockerfile/internals_test.go
+++ b/engine/builder/dockerfile/internals_test.go
@@ -16,8 +16,9 @@
 	"github.com/stretchr/testify/require"
 )
 
 func TestEmptyDockerfile(t *testing.T) {
+	t.Skip("DM - skipping privileged test")
 	contextDir, cleanup := createTestTempDir(t, "", "builder-dockerfile-test")
 	defer cleanup()
 
 	createTestTempFile(t, contextDir, builder.DefaultDockerfileName, "", 0777)
@@ -25,8 +26,9 @@
 	readAndCheckDockerfile(t, "emptyDockerfile", contextDir, "", "the Dockerfile (Dockerfile) cannot be empty")
 }
 
 func TestSymlinkDockerfile(t *testing.T) {
+	t.Skip("DM - skipping privileged test")
 	contextDir, cleanup := createTestTempDir(t, "", "builder-dockerfile-test")
 	defer cleanup()
 
 	createTestSymlink(t, contextDir, builder.DefaultDockerfileName, "/etc/passwd")
@@ -40,8 +42,9 @@
 	readAndCheckDockerfile(t, "symlinkDockerfile", contextDir, builder.DefaultDockerfileName, expectedError)
 }
 
 func TestDockerfileOutsideTheBuildContext(t *testing.T) {
+	t.Skip("DM - skipping privileged test")
 	contextDir, cleanup := createTestTempDir(t, "", "builder-dockerfile-test")
 	defer cleanup()
 
 	expectedError := "Forbidden path outside the build context: ../../Dockerfile ()"
@@ -49,8 +52,9 @@
 	readAndCheckDockerfile(t, "DockerfileOutsideTheBuildContext", contextDir, "../../Dockerfile", expectedError)
 }
 
 func TestNonExistingDockerfile(t *testing.T) {
+	t.Skip("DM - skipping privileged test")
 	contextDir, cleanup := createTestTempDir(t, "", "builder-dockerfile-test")
 	defer cleanup()
 
 	expectedError := "Cannot locate specified Dockerfile: Dockerfile"
--- a/engine/daemon/daemon_test.go
+++ b/engine/daemon/daemon_test.go
@@ -151,8 +151,9 @@
 	}
 }
 
 func TestContainerInitDNS(t *testing.T) {
+	t.Skip("DM - skipping privileged test")
 	tmp, err := ioutil.TempDir("", "docker-container-test-")
 	if err != nil {
 		t.Fatal(err)
 	}
--- a/engine/daemon/daemon_unix_test.go
+++ b/engine/daemon/daemon_unix_test.go
@@ -274,8 +274,9 @@
 	}
 }
 
 func TestMigratePre17Volumes(t *testing.T) {
+	t.Skip("DM - Skipping privileged test")
 	rootDir, err := ioutil.TempDir("", "test-daemon-volumes")
 	if err != nil {
 		t.Fatal(err)
 	}
--- a/engine/daemon/graphdriver/overlay2/overlay_test.go
+++ b/engine/daemon/graphdriver/overlay2/overlay_test.go
@@ -48,24 +48,29 @@
 
 // This avoids creating a new driver for each test if all tests are run
 // Make sure to put new tests between TestOverlaySetup and TestOverlayTeardown
 func TestOverlaySetup(t *testing.T) {
+	t.Skip("DM - skipping privileged test")
 	graphtest.GetDriver(t, driverName)
 }
 
 func TestOverlayCreateEmpty(t *testing.T) {
+	t.Skip("DM - skipping privileged test")
 	graphtest.DriverTestCreateEmpty(t, driverName)
 }
 
 func TestOverlayCreateBase(t *testing.T) {
+	t.Skip("DM - skipping privileged test")
 	graphtest.DriverTestCreateBase(t, driverName)
 }
 
 func TestOverlayCreateSnap(t *testing.T) {
+	t.Skip("DM - skipping privileged test")
 	graphtest.DriverTestCreateSnap(t, driverName)
 }
 
 func TestOverlay128LayerRead(t *testing.T) {
+	t.Skip("DM - skipping privileged test")
 	graphtest.DriverTestDeepLayerRead(t, 128, driverName)
 }
 
 func TestOverlayDiffApply10Files(t *testing.T) {
--- a/engine/volume/local/local_test.go
+++ b/engine/volume/local/local_test.go
@@ -29,8 +29,9 @@
 
 }
 
 func TestRemove(t *testing.T) {
+	t.Skip("DM - skipping privileged tests")
 	// TODO Windows: Investigate why this test fails on Windows under CI
 	//               but passes locally.
 	if runtime.GOOS == "windows" {
 		t.Skip("Test failing on Windows CI")
@@ -76,8 +77,9 @@
 	}
 }
 
 func TestInitializeWithVolumes(t *testing.T) {
+	t.Skip("DM - skipping privileged tests")
 	rootDir, err := ioutil.TempDir("", "local-volume-test")
 	if err != nil {
 		t.Fatal(err)
 	}
@@ -108,8 +110,9 @@
 	}
 }
 
 func TestCreate(t *testing.T) {
+	t.Skip("DM - skipping privileged tests")
 	rootDir, err := ioutil.TempDir("", "local-volume-test")
 	if err != nil {
 		t.Fatal(err)
 	}
@@ -180,8 +183,9 @@
 	}
 }
 
 func TestCreateWithOpts(t *testing.T) {
+t.Skip("DM - skipping privileged test")
 	if runtime.GOOS == "windows" {
 		t.Skip()
 	}
 	rootDir, err := ioutil.TempDir("", "local-volume-test")
@@ -286,8 +290,9 @@
 	}
 }
 
 func TestRealodNoOpts(t *testing.T) {
+	t.Skip("DM - skipping privileged tests")
 	rootDir, err := ioutil.TempDir("", "volume-test-reload-no-opts")
 	if err != nil {
 		t.Fatal(err)
 	}
--- a/engine/pkg/mount/mount_unix_test.go
+++ b/engine/pkg/mount/mount_unix_test.go
@@ -24,8 +24,9 @@
 	}
 }
 
 func TestMounted(t *testing.T) {
+	t.Skip("DM - skip privileged test")
 	tmp := path.Join(os.TempDir(), "mount-tests")
 	if err := os.MkdirAll(tmp, 0777); err != nil {
 		t.Fatal(err)
 	}
@@ -75,8 +76,9 @@
 	}
 }
 
 func TestMountReadonly(t *testing.T) {
+	t.Skip("DM - skip privileged test")
 	tmp := path.Join(os.TempDir(), "mount-tests")
 	if err := os.MkdirAll(tmp, 0777); err != nil {
 		t.Fatal(err)
 	}
@@ -120,8 +122,9 @@
 	}
 }
 
 func TestGetMounts(t *testing.T) {
+	t.Skip("DM - skip privileged test")
 	mounts, err := GetMounts()
 	if err != nil {
 		t.Fatal(err)
 	}
--- a/engine/pkg/mount/sharedsubtree_linux_test.go
+++ b/engine/pkg/mount/sharedsubtree_linux_test.go
@@ -11,8 +11,9 @@
 )
 
 // nothing is propagated in or out
 func TestSubtreePrivate(t *testing.T) {
+	t.Skip("DM - skip privileged test")
 	tmp := path.Join(os.TempDir(), "mount-tests")
 	if err := os.MkdirAll(tmp, 0777); err != nil {
 		t.Fatal(err)
 	}
@@ -109,8 +110,9 @@
 
 // Testing that when a target is a shared mount,
 // then child mounts propagate to the source
 func TestSubtreeShared(t *testing.T) {
+	t.Skip("DM - skip privileged test")
 	tmp := path.Join(os.TempDir(), "mount-tests")
 	if err := os.MkdirAll(tmp, 0777); err != nil {
 		t.Fatal(err)
 	}
@@ -177,8 +179,9 @@
 
 // testing that mounts to a shared source show up in the slave target,
 // and that mounts into a slave target do _not_ show up in the shared source
 func TestSubtreeSharedSlave(t *testing.T) {
+	t.Skip("DM - skip privileged test")
 	tmp := path.Join(os.TempDir(), "mount-tests")
 	if err := os.MkdirAll(tmp, 0777); err != nil {
 		t.Fatal(err)
 	}
@@ -281,8 +284,9 @@
 	}
 }
 
 func TestSubtreeUnbindable(t *testing.T) {
+	t.Skip("DM - skip privileged test")
 	tmp := path.Join(os.TempDir(), "mount-tests")
 	if err := os.MkdirAll(tmp, 0777); err != nil {
 		t.Fatal(err)
 	}
--- a/engine/pkg/chrootarchive/archive_test.go
+++ b/engine/pkg/chrootarchive/archive_test.go
@@ -40,8 +40,9 @@
 	return chrootArchiver.CopyWithTar(src, dst)
 }
 
 func TestChrootTarUntar(t *testing.T) {
+	t.Skip("DM - skip privileged test")
 	tmpdir, err := ioutil.TempDir("", "docker-TestChrootTarUntar")
 	if err != nil {
 		t.Fatal(err)
 	}
@@ -71,8 +72,9 @@
 
 // gh#10426: Verify the fix for having a huge excludes list (like on `docker load` with large # of
 // local images)
 func TestChrootUntarWithHugeExcludesList(t *testing.T) {
+	t.Skip("DM - skip privileged test")
 	tmpdir, err := ioutil.TempDir("", "docker-TestChrootUntarHugeExcludes")
 	if err != nil {
 		t.Fatal(err)
 	}
@@ -169,8 +171,9 @@
 	return nil
 }
 
 func TestChrootTarUntarWithSymlink(t *testing.T) {
+	t.Skip("DM - skip privileged test")
 	// TODO Windows: Figure out why this is failing
 	if runtime.GOOS == "windows" {
 		t.Skip("Failing on Windows")
 	}
@@ -195,8 +198,9 @@
 	}
 }
 
 func TestChrootCopyWithTar(t *testing.T) {
+t.Skip("DM - skipping privileged test")
 	// TODO Windows: Figure out why this is failing
 	if runtime.GOOS == "windows" {
 		t.Skip("Failing on Windows")
 	}
@@ -245,8 +249,9 @@
 	}
 }
 
 func TestChrootCopyFileWithTar(t *testing.T) {
+	t.Skip("DM - skip privileged test")
 	tmpdir, err := ioutil.TempDir("", "docker-TestChrootCopyFileWithTar")
 	if err != nil {
 		t.Fatal(err)
 	}
@@ -288,8 +293,9 @@
 	}
 }
 
 func TestChrootUntarPath(t *testing.T) {
+	t.Skip("DM - skip privileged test")
 	// TODO Windows: Figure out why this is failing
 	if runtime.GOOS == "windows" {
 		t.Skip("Failing on Windows")
 	}
@@ -353,8 +359,9 @@
 	return count, nil
 }
 
 func TestChrootUntarEmptyArchiveFromSlowReader(t *testing.T) {
+	t.Skip("DM - skip privileged test")
 	tmpdir, err := ioutil.TempDir("", "docker-TestChrootUntarEmptyArchiveFromSlowReader")
 	if err != nil {
 		t.Fatal(err)
 	}
@@ -369,8 +376,9 @@
 	}
 }
 
 func TestChrootApplyEmptyArchiveFromSlowReader(t *testing.T) {
+	t.Skip("DM - skip privileged test")
 	tmpdir, err := ioutil.TempDir("", "docker-TestChrootApplyEmptyArchiveFromSlowReader")
 	if err != nil {
 		t.Fatal(err)
 	}
@@ -385,8 +393,9 @@
 	}
 }
 
 func TestChrootApplyDotDotFile(t *testing.T) {
+	t.Skip("DM - skip privileged test")
 	tmpdir, err := ioutil.TempDir("", "docker-TestChrootApplyDotDotFile")
 	if err != nil {
 		t.Fatal(err)
 	}
--- a/engine/daemon/graphdriver/overlay/overlay_test.go
+++ b/engine/daemon/graphdriver/overlay/overlay_test.go
@@ -18,24 +18,29 @@
 
 // This avoids creating a new driver for each test if all tests are run
 // Make sure to put new tests between TestOverlaySetup and TestOverlayTeardown
 func TestOverlaySetup(t *testing.T) {
+	t.Skip("DM - skipping privileged test")
 	graphtest.GetDriver(t, "overlay")
 }
 
 func TestOverlayCreateEmpty(t *testing.T) {
+	t.Skip("DM - skipping privileged test")
 	graphtest.DriverTestCreateEmpty(t, "overlay")
 }
 
 func TestOverlayCreateBase(t *testing.T) {
+	t.Skip("DM - skipping privileged test")
 	graphtest.DriverTestCreateBase(t, "overlay")
 }
 
 func TestOverlayCreateSnap(t *testing.T) {
+	t.Skip("DM - skipping privileged test")
 	graphtest.DriverTestCreateSnap(t, "overlay")
 }
 
 func TestOverlay50LayerRead(t *testing.T) {
+	t.Skip("DM - skipping privileged test")
 	graphtest.DriverTestDeepLayerRead(t, 50, "overlay")
 }
 
 // Fails due to bug in calculating changes after apply
--- a/engine/pkg/archive/archive_test.go
+++ b/engine/pkg/archive/archive_test.go
@@ -428,8 +428,9 @@
 	}
 }
 
 func TestCopyWithTarInexistentDestWillCreateIt(t *testing.T) {
+	t.Skip("DM - skipping privileged test")
 	tempFolder, err := ioutil.TempDir("", "docker-archive-test")
 	if err != nil {
 		t.Fatal(nil)
 	}
@@ -961,8 +962,9 @@
 	}
 }
 
 func TestUntarHardlinkToSymlink(t *testing.T) {
+	t.Skip("DM - skipping privileged test")
 	// TODO Windows. There may be a way of running this, but turning off for now
 	if runtime.GOOS == "windows" {
 		t.Skip("hardlinks on Windows")
 	}
--- a/engine/pkg/system/rm_test.go
+++ b/engine/pkg/system/rm_test.go
@@ -39,8 +39,9 @@
 	}
 }
 
 func TestEnsureRemoveAllWithMount(t *testing.T) {
+t.Skip("DM - skipping privileged test")
 	if runtime.GOOS == "windows" {
 		t.Skip("mount not supported on Windows")
 	}
 
--- a/engine/daemon/reload_test.go
+++ b/engine/daemon/reload_test.go
@@ -481,8 +481,9 @@
 	}
 }
 
 func TestDaemonReloadNetworkDiagnosticPort(t *testing.T) {
+t.Skip("DM - skipping privileged test")
 	daemon := &Daemon{}
 	daemon.configStore = &config.Config{}
 
 	valuesSet := make(map[string]interface{})
